docs/model-dev-guide/batch-processing/batch-process-api-ug.rst

```diff
@@ -0,0 +1,173 @@
+:class:`~determined.pytorch.MetricReducer`. You can pass MetricReducer to
+:class:`~determined.pytorch.experimental.TorchBatchProcessor` as follow:
+
+``TorchBatchProcessorContext``
+==============================
+
+:class:`~determined.pytorch.experimental.TorchBatchProcessorContext` should be a subclass of
+:class:`~determined.pytorch.experimental.TorchBatchProcessor`. The two functions you must implement
+are the :meth:`~determined.pytorch.experimental.TorchBatchProcessor.__init__` and
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.process_batch`. The other lifecycle
+functions are optional.
+
+.. code:: python
+
+   class MyProcessor(TorchBatchProcessor):
+       def __init__(self, context):
+           self.reducer = context.wrap_reducer(reducer=AccuracyMetricReducer(), name="accuracy")
+
+******************************************
+ How To Perform Batch (Offline) Inference
+******************************************
+
+In this section, we'll learn how to perform batch inference using the Torch Batch Processing API.
+
+Step 1: Define a InferenceProcessor
+===================================
+
+The first step is to define an InferenceProcessor. You should initialize your model in the
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.__init__` function of the
+InferenceProcessor. You should implement
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.process_batch` function with inference
+logic.
+
+You can optionally implement
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.on_checkpoint_start` and
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.on_finish` function to be run before
+every checkpoint and after all the data are processed respectively.
+
+.. code:: python
+
+   """
+   Define custom processor class
+   """
+   class InferenceProcessor(TorchBatchProcessor):
+       def __init__(self, context):
+           self.context = context
+           self.model = context.prepare_model_for_inference(get_model())
+           self.output = []
+           self.last_index = 0
+
+       def process_batch(self, batch, batch_idx) -> None:
+           model_input = batch[0]
+           model_input = self.context.to_device(model_input)
+
+           with torch.no_grad():
+               with self.profiler as p:
+                   pred = self.model(model_input)
+                   p.step()
+                   output = {"predictions": pred, "input": batch}
+                   self.output.append(output)
+
+           self.last_index = batch_idx
+
+       def on_checkpoint_start(self):
+           """
+           During checkpoint, we persist prediction result
+           """
+           if len(self.output) == 0:
+               return
+           file_name = f"prediction_output_{self.last_index}"
+           with self.context.upload_path() as path:
+               file_path = pathlib.Path(path, file_name)
+               torch.save(self.output, file_path)
+
+           self.output = []
+
```

Comment:
If this is straight from the code in the repo, then we can use "literalinclude", e.g.,

.. literalinclude:: ../../../directoryl/filename.py
   :language: python
   :start-after: # Define custom processor class
   :end-after: # self.output = []
   :dedent:

even better if we add the Docs snippet comments to the code, e.g., # Docs snippet start: custom processor class, # Docs snippet end: custom processor class

Reply:
Discussed offline, leave as is for now.